Add smart_proxy_shellhooks (RPM)

lzap opened the pull request:
For plugin updates, please indicate which repos this should be built into:

* Nightly
* 2.3

Do not merge yet, need to test the build.

evgeni commented:
```
error: Installed (but unpackaged) file(s) found:
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/README
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/ansible-tower-cli
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/print_args
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/print_body
```

evgeni commented:
```
error: Installed (but unpackaged) file(s) found:
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/README
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/ansible-tower-cli
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/print_args
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/print_body
   /var/lib/rubygem-smart_proxy_shellhooks/shellhooks/README
   /var/lib/rubygem-smart_proxy_shellhooks/shellhooks/ansible-tower-cli
   /var/lib/rubygem-smart_proxy_shellhooks/shellhooks/print_args
   /var/lib/rubygem-smart_proxy_shellhooks/shellhooks/print_body
```

`/var/lib` sounds wrong to me?

lzap commented:
Do not merge, still testing.

lzap commented:
Now it finally works:

```[root@katello ~]# rpm -ql tfm-rubygem-smart_proxy_shellhooks
/etc/foreman-proxy/settings.d/shellhooks.yml
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/LICENSE
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks/shellhooks.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks/shellhooks_api.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks/shellhooks_http_config.ru
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks/version.rb
/opt/theforeman/tfm/root/usr/share/gems/specifications/smart_proxy_shellhooks-0.9.0.gemspec
/usr/share/foreman-proxy/bundler.d/shellhooks.rb
/var/lib/foreman-proxy/shellhooks
/var/lib/foreman-proxy/shellhooks/README
/var/lib/foreman-proxy/shellhooks/ansible-tower-cli
/var/lib/foreman-proxy/shellhooks/print_args
/var/lib/foreman-proxy/shellhooks/print_body

[root@katello ~]# curl -k https://localhost:9090/shellhook/
{"valid":["ansible-tower-cli","print_args","print_body"],"invalid":["README"]}
```

lzap commented:
I am gonna push 0.9.1 out and rebuild once again.

lzap commented:
It works fine now. RTM

lzap commented:
Hello, anyone there? I will be working on Debian package this week.

evgeni commented on packages/plugins/rubygem-smart_proxy_shellhooks/rubygem-smart_proxy_shellhooks.spec:
```diff
@@ -0,0 +1,118 @@
+Provides: foreman-proxy-plugin-%{plugin_name} = %{version}
+# end specfile generated dependencies
+
+%{?scl:Obsoletes: rubygem-%{gem_name}}
```
you can drop that, we never had a rubygm-shellhooks outside the scl.

lzap replied:
Dropped

evgeni commented:
[test rpm]

While not packaging related, I find the "invalid: README" weird -- should that be special-handled in the plugin?

evgeni reviewed (approved):
the `Obsoletes` is not needed, but also doesn't hurt.

ACK, please ping if you want to fix the obsoletes.

lzap commented:
I dropped the line, rebased. Thanks.

> While not packaging related, I find the "invalid: README" weird -- should that be special-handled in the plugin?

Since `README` is not executable, it's reported as an invalid shell hook. Not sure what exactly do you find weird. This endpoint is never called like that, this is more like a debugging endpoint. Foreman only does `POST` here.

evgeni commented:
you don't seem to actually have pushed the changes :)
